lib/shanty_docker_plugin/docker_credentials.rb

```diff
@@ -0,0 +1,47 @@
+require 'base64'
+require 'docker'
+require 'json'
+require 'uri'
+
+module DockerPlugin
+  # Public: Set up credentials from ~/.dockercfg
+  class DockerCredentials
+    def initialize(dockercfg_file = "#{Dir.home}/.dockercfg")
+      @dockercfg_file = dockercfg_file
+    end
+
+    def auth!(registry)
+      Docker.creds = find_credentials(registry)
+    end
+
+    private
+
+    def find_credentials(registry)
+      credentials[registry] || credentials['default'] || {}
+    end
+
+    def credentials
+      @credentials ||= read_dockercfg.each_with_object({}) do |v, acc|
+        username, password = decode_credentials(v.last['auth'])
+        acc[registry(v.first)] = { username: username, password: password, email: v.last['email'] }
+      end
+    end
+
+    def registry(url)
+      host = url
+      host = URI(url).host if url.start_with?('http')
+      return 'default' if host.end_with?('docker.io')
```

Comment:
Might be able to do something there. Your host *does* end with docker.io, as it's a URI:

```ruby
host = URI(url).host if url.start_with?('http')
```